omod/src/main/java/org/openmrs/module/reporting/web/reports/ManageReportsController.java

```diff
@@ -77,6 +77,20 @@ public ModelMap manageReports(ModelMap model,
     	// Get possible new reports to create
     	Map<String, String> types = new LinkedHashMap<String, String>();
     	types.put("Period Indicator Report", "periodIndicatorReport.form");
+	try {
+            Context c = new Context(); 
+            Class ls = c.loadClass("LogicService");
+            if ("LoigcService".equals(ls.getName())){
+                //types.put("Row-Per-Patient Report", "logicReport.form");   
+                System.out.println("logc report service here");
+            }
+            else {
+                System.out.println("*******LogicService is installed*******");
+            }
+
+        } catch (ClassNotFoundException e){
+            System.err.println(e.getMessage());
+        }
     	types.put("Row-Per-Patient Report", "logicReport.form");
```

Comment:
@TheFakeTedCruz , the approach is not one we would want to take.  You can check ModuleFactory.isModuleStarted to determine if the logic module is installed and running.